src/resource/solidDataset.ts

```diff
@@ -1156,14 +1156,41 @@ function resolveLocalIrisInThing(
 }
 
 /**
- * Fetch the contents of '.well-known/solid' for a given resource URL.
+ * Fetch a SolidDataset containing information about the capabilities of the
+ * storage server that hosts the given resource URL. For more information,
+ * please see the [ESS
+ * Documentation](https://docs.inrupt.com/ess/latest/services/discovery-endpoint/#well-known-solid).
+ *
+ * **Note:** The data contained in this dataset has changed between ESS 1.1 and
+ * ESS 2.0, as such you will need to check for multiple predicates to support
+ * both versions.
+ *
+ * ```typescript
+ * const wellKnown = await getWellKnownSolid(resource);
+ *
+ * // The wellKnown dataset uses a blank node for the subject all of it’s predicates,
+ * // such that we need to call getThingAll with acceptBlankNodes set to true to
+ * // retrieve back predicates contained within the dataset
+ * const wellKnownSubjects = getThingAll(wellKnown, {
+ *   acceptBlankNodes: true,
+ * });
+ * const wellKnownSubject = wellKnownSubjects[0];
```

Comment:
What we're doing here is getting all subjects in the document, and getting the first. But the document might contain anything, and RDF defines no fixed order. So `[0]` could point to absolutely anything.

Reply:
We're pretty much expecting this to only have one subject in the document — if we fail to get a value, then we just disable that functionality

Reply:
That's a very strong assumption, which breaka for instanstance when any value in the JSON structure becomes composite.

(Note that all my remarks are about longterm sustainable coding with RDF data; if that's not a requirement then we're good, but we might be better off reading the JSON.)

Reply:
Yup, let's address this by getting spec to cover the reason we have `getWellKnownSolid` — e.g., a `getStorageMetadata(iri: string | URL): Record<string, string | string[]>` or something.